### PR TITLE
Add support for windows CMake builds

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -405,6 +405,13 @@ AC_DEFUN_ONCE([CUSTOM_LATE_HOOK],
 
   # explicitly disable classlist generation
   ENABLE_GENERATE_CLASSLIST=false
+
+  if test "x$OPENJDK_BUILD_OS" = xwindows ; then
+    OPENJ9_TOOL_DIR="$OUTPUTDIR/tools"
+    AC_SUBST([OPENJ9_TOOL_DIR])
+    OPENJ9_GENERATE_TOOL_WRAPPERS
```
Comment (pshipton) [inline closed/autoconf/custom-hook.m4]: I don't see anything making this specific to cmake builds.

Comment (dnakamura) [reply closed/autoconf/custom-hook.m4]: I can change it if you want. I figured it really didn't matter if we generate a few text files that potentially go unused

**Comment by pshipton** [comment]
jenkins test sanity win

File: closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -540,3 +547,36 @@ AC_DEFUN([CONFIGURE_OPENSSL],
   AC_SUBST(BUILD_OPENSSL)
   AC_SUBST(OPENSSL_CFLAGS)
 ])
+
+# Create a tool wrapper for use by cmake.
+# Consists of a shell script which wraps commands with an invocation of fixpath.
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: Why do we need this? Why can't we simply ask cmake to use `CC=fixpath cl ...`?

Comment (dnakamura) [reply closed/autoconf/custom-hook.m4]: It appears to work the first time, but when rebuilding,  sometimes it just stops working. I believe it occurs when cmake decides it needs to re-run itself and needs to re-test the compilers.

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: Wouldn't it be better (and perhaps simpler) to just fix that?

Comment (dnakamura) [reply closed/autoconf/custom-hook.m4]: no. It is a limitiation/bug in cmake itself

File: closed/OpenJ9.gmk [hunk 3]
```diff
@@ -454,22 +454,26 @@ ifneq (,$(OPENJ9_DEVELOPER_DIR))
...
+    else
+      CMAKE_ARGS += "-DCMAKE_C_COMPILER=$(ac_cv_prog_CC)"
+    endif
+
+    ifneq (,$(OPENJ9_CXX))
+      CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(OPENJ9_CXX)"
+    else
+      CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(ac_cv_prog_CXX)"
+    endif
+endif # OPENJ9_ENABLE_CMAKE
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: This `endif` is for Windows, not CMake; and is not properly indented.

File: closed/autoconf/custom-hook.m4 (hunk 2 above)
Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: I'm not convinced: I think we should explore this further (later).

File: closed/autoconf/custom-hook.m4 [hunk 4]
```diff
@@ -579,3 +586,35 @@ AC_DEFUN([CONFIGURE_OPENSSL],
...
+# OPENJ9_GENERATE_TOOL_WRAPER(<name_of_wrapper>, <command_to_call>)
+AC_DEFUN([OPENJ9_GENERATE_TOOL_WRAPPER],
+[
+  tool_file="$OPENJ9_TOOL_DIR/$1"
+
+  echo "#!/bin/sh" > $tool_file
+  printf '%s "%s" "$[]@"\n' "$FIXPATH" "$2" >> $tool_file
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: I think this deserves a comment that explains the use of `[]` in the format string.

File: closed/autoconf/custom-hook.m4 [hunk 5]
```diff
@@ -579,3 +586,37 @@ AC_DEFUN([CONFIGURE_OPENSSL],
...
+# Consists of a shell script which wraps commands with an invocation of fixpath.
+# OPENJ9_GENERATE_TOOL_WRAPER(<name_of_wrapper>, <command_to_call>)
+AC_DEFUN([OPENJ9_GENERATE_TOOL_WRAPPER],
+[
+  tool_file="$OPENJ9_TOOL_DIR/$1"
+
+  echo "#!/bin/sh" > $tool_file
+  # We need to insert an empty string ([]), to stop M4 treating "$@" as a
+  # variable reference
+  printf '%s "%s" "$[]@"\n' "$FIXPATH" "$2" >> $tool_file
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: It's actually `/bin/sh` that we're talking to here, and now that the control string is enclosed in single-quotes, I don't think `[]` is necessary.

**Comment by keithc-ca** [comment]
Jenkins test sanity win,win32,zlinux jdk11

**Pull request merged**